micro-deps/micro-deps-jax-rs/src/main/java/com/ofg/infrastructure/jaxrs/JaxRsServiceResolver.java

```diff
@@ -0,0 +1,17 @@
+package com.ofg.infrastructure.jaxrs;
+
+import com.google.common.base.Supplier;
+
+/**
+ * Allows to obtain JAX-RS proxy service interface implementation from its interface
+ */
+public interface JaxRsServiceResolver {
```

Comment:
This is currently handled by placing the supplier call close enought to the actual call. We can implement your proposal later as extra method, but there are situations when caller needs to make sure that multiple calls in sequence are being executed on the same remote service instance, and even if it goes down it is less of  an issue, than if the call lands on a different instance. So the supplier implementation is still required.